### PR TITLE
questionnaire content value type changed to Any from String

### PR DESCRIPTION
**Pull request by siddh1004**
https://app.shortcut.com/simpledotorg/story/10087/change-implementation-of-content-map-in-questionnaireresponse

### CODE REVIEW

**Comment by siddh1004** [comment]
1. `OrientationType` removed from the display properties of questionnaire form layout response
2. Type of content changed to Map<String, Any> from Map<String, String>
3. QuestionnaireResponse property `last_updated_by_userid` changed to nullable.

File: app/src/main/java/org/simple/clinic/storage/migrations/RoomMigrationsModule.kt [hunk 1]
```diff
@@ -113,7 +113,8 @@ class RoomMigrationsModule {
       migration105: Migration_105,
       migration106: Migration_106,
       migration107: Migration_107,
-      migration108: Migration_108
+      migration108: Migration_108,
+      migration109: Migration_109
```
Comment (olttwa) [inline app/src/main/java/org/simple/clinic/storage/migrations/RoomMigrationsModule.kt:117]:
```suggestion
      migration109: Migration_109,
```

Comment (olttwa) [reply app/src/main/java/org/simple/clinic/storage/migrations/RoomMigrationsModule.kt:117]: Kotlin accepts comma in the last element of a list. I suggest adding a comma always so it isn't included in a diff next time an entry is added.

File: app/src/main/java/org/simple/clinic/questionnaireresponse/QuestionnaireResponse.kt [hunk 2]
```diff
@@ -29,9 +30,9 @@ data class QuestionnaireResponse(
 
     val facilityId: UUID,
 
-    val lastUpdatedByUserId: UUID,
+    val lastUpdatedByUserId: UUID?,
 
-    val content: Map<String, String>,
+    val content: Map<String, @RawValue Any>,
```
Comment (olttwa) [inline app/src/main/java/org/simple/clinic/questionnaireresponse/QuestionnaireResponse.kt:35]: This was a good find. Congrats 🎉

File: app/src/androidTest/java/org/simple/clinic/sync/QuestionnaireResponseSyncIntegrationTest.kt [hunk 3]
```diff
@@ -91,32 +75,18 @@ class QuestionnaireResponseSyncIntegrationTest {
...
   }
 
   private fun resetLocalData() {
-    clearQuestionnaireData()
     clearQuestionnaireResponseData()
     lastPullToken.delete()
-    questionnaireLastPullToken.delete()
-  }
-
-  private fun clearQuestionnaireData() {
```
Comment (olttwa) [inline app/src/androidTest/java/org/simple/clinic/sync/QuestionnaireResponseSyncIntegrationTest.kt:118]: I see you've cleaned up functions which weren't of any use as of this moment. That's good. We'll add them later if needed.

**Review by olttwa — DISMISSED** [review]

File: app/src/main/java/org/simple/clinic/storage/migrations/RoomMigrationsModule.kt (hunk 1 above)
Comment (olttwa) [reply app/src/main/java/org/simple/clinic/storage/migrations/RoomMigrationsModule.kt:117]: For example, if `Migration_108,` had a trailing comma, it wouldn't show up in this diff

File: app/schemas/org.simple.clinic.AppDatabase/109.json [hunk 4]
```diff
@@ -0,0 +1,2143 @@
+{
```
Comment (prabhanshuguptagit) [inline app/schemas/org.simple.clinic.AppDatabase/109.json:1]: Does this file need to be committed?

**Review by prabhanshuguptagit — CHANGES_REQUESTED** [review]
Looks good, one question about committing the AppDatabase json file.

File: app/schemas/org.simple.clinic.AppDatabase/109.json (hunk 4 above)
Comment (siddh1004) [reply app/schemas/org.simple.clinic.AppDatabase/109.json:1]: yes

Comment (siddh1004) [reply app/schemas/org.simple.clinic.AppDatabase/109.json:1]: @prabhanshuguptagit yes

**Review by prabhanshuguptagit — APPROVED** [review]

**Pull request merged**